src/util/session.ts

```diff
@@ -32,9 +36,9 @@ export function resolveVariable(key: string) {
 /**
  * Prints formatted text to stdout, supports placeholders and automatically stringifies objects
  * @param str Format string
- * @param args Format args. If str has no format placeholders these get appended to str separated by newlines
+ * @param [args] Format args. If str has no format placeholders these get appended to str separated by newlines
  */
-export function printf(str: string, ...args: unknown[]) {
+export function printf<T>(str: string, ...args: T[]) {
```

Comment:
what is this for? this doesn't allow passing in mixed types, or does it?

Reply:
it does